### ISSUE TITLE
Highlight search results

### ISSUE THREAD

**Issue by antoyo** [helix-editor/helix#1733]
Hi.

#### Describe your feature request

I'd like an option to highlight the search results.
Thanks.

**Comment by kirawi** [comment]
I think this functionality was recently merged with https://github.com/helix-editor/helix/pull/1635

**Comment by antoyo** [comment]
That seems to only highlight the search results in the file picker.

I was talking about search in the buffer.

**Comment by archseer** [comment]
Search is selection based though, a way to see all the results would be to use `vnnnnn` to extend the selection with all matches then `(` and `)` to cycle the primary selection

**Comment by archseer** [comment]
You can also `%s` followed by search query to select all the matches.

**Comment by antoyo** [comment]
That would produce a different behavior, though.

[Kakoune supports it.](https://github.com/mawww/kakoune/commit/25451ac1128feb0f1ce8c1adbf327964f6b8b086) For instance, you can create a search highlighter this way:
```
:addhl buffer/ dynregex '%reg{/}' 0:+u 1:+b
```

**Comment by xJonathanLEI** [comment]
Just high-lighting all the search result is not exactly the same as selecting all of them (and hence... highlighting) though. I would like to loop through all the matches, and once I find the match I'm interested in, stop and make changes (ONLY to the current match), and then continue.

Having the next and previous matches highlighted helps me quickly navigate, as I can have a reasonable guess on where I would land before I even hit `n` (am I gonna jump to another function? or really just next line?). It saves a lot of mental effort trying to identify where I am, especially with the lack of smooth scrolling in terminal.

**Comment by cemalokten** [comment]
> Just high-lighting all the search result is not exactly the same as selecting all of them (and hence... highlighting) though. I would like to loop through all the matches, and once I find the match I'm interested in, stop and make changes (ONLY to the current match), and then continue.
> 
> Having the next and previous matches highlighted helps me quickly navigate, as I can have a reasonable guess on where I would land before I even hit `n` (am I gonna jump to another function? or really just next line?). It saves a lot of mental effort trying to identify where I am, especially with the lack of smooth scrolling in terminal.

I totally agree, the ability to highlight all search results in the buffer would be incredibly useful, something I used to do in neovim all the time before switching to Helix.

**Comment by eguiraud** [comment]
Maybe easier now that virtual text #411 has been merged?
